homeassistant/components/switchbot_cloud/vacuum.py

```diff
@@ -110,10 +137,78 @@ def _set_attributes(self) -> None:
         switchbot_state = str(self.coordinator.data.get("workingStatus"))
         self._attr_activity = VACUUM_SWITCHBOT_STATE_TO_HA_STATE.get(switchbot_state)
 
+        if self._attr_fan_speed is None:
+            self._attr_fan_speed = VACUUM_FAN_SPEED_QUIET
+
+
+class SwitchBotCloudVacuumV2(SwitchBotCloudVacuum):
```

Comment:
Same here. The library should have some sort of capabilities, which than uses the correct command...
We don't want the logic for different vacuum models direct in HA

Reply:
done